plugins/catalog-backend/src/ingestion/processors/GithubMultiOrgReaderProcessor.ts

```diff
@@ -0,0 +1,181 @@
+    logger: Logger;
+    orgs: GithubMultiOrgConfig;
+  }) {
+    this.integrations = options.integrations;
+    this.logger = options.logger;
+    this.orgs = options.orgs;
+  }
+
+  async readLocation(
+    location: LocationSpec,
+    _optional: boolean,
+    emit: CatalogProcessorEmit,
+  ): Promise<boolean> {
+    if (location.type !== 'github-multi-org') {
+      return false;
+    }
+
+    const gitHubConfig = this.integrations.github.byUrl(location.target)
+      ?.config;
+    if (!gitHubConfig) {
+      throw new Error(
+        `There is no GitHub integration that matches ${location.target}. Please add a configuration entry for it under integrations.github`,
+      );
+    }
+
+    const allUsersMap = new Map();
+    const baseUrl = new URL(location.target).origin;
+    const credentialsProvider = GithubCredentialsProvider.create(gitHubConfig);
+
+    const orgsToProcess = this.orgs.length
+      ? this.orgs
+      : await this.getAllOrgs(gitHubConfig);
+
+    for (const orgConfig of orgsToProcess) {
+      try {
+        const {
+          headers,
+          type: tokenType,
+        } = await credentialsProvider.getCredentials({
+          url: `${baseUrl}/${orgConfig.name}`,
+        });
+        const client = graphql.defaults({
+          baseUrl: gitHubConfig.apiBaseUrl,
+          headers,
+        });
+
+        const startTimestamp = Date.now();
+        this.logger.info(
+          `Reading GitHub users and teams for org: ${orgConfig.name}`,
+        );
+        const { users } = await getOrganizationUsers(
+          client,
+          orgConfig.name,
+          tokenType,
+        );
+        const { groups, groupMemberUsers } = await getOrganizationTeams(
+          client,
+          orgConfig.name,
+          orgConfig.groupNamespace,
+        );
+
+        const duration = ((Date.now() - startTimestamp) / 1000).toFixed(1);
+        this.logger.debug(
+          `Read ${users.length} GitHub users and ${groups.length} GitHub teams from ${orgConfig.name} in ${duration} seconds`,
+        );
+
+        users.forEach(u => {
+          if (!allUsersMap.has(u.metadata.name)) {
+            allUsersMap.set(u.metadata.name, u);
+          }
+        });
+
+        for (const [groupName, userNames] of groupMemberUsers.entries()) {
+          for (const userName of userNames) {
+            const user = allUsersMap.get(userName);
+            if (user && !user.spec.memberOf.includes(groupName)) {
+              user.spec.memberOf.push(groupName);
+            }
+          }
+        }
+        buildOrgHierarchy(groups);
+
+        for (const group of groups) {
+          emit(results.entity(location, group));
+        }
+      } catch (e) {
+        this.logger.error(
+          `Failed to read GitHub org data for ${orgConfig.name}: ${e}`,
+        );
+      }
+    }
+
+    const allUsers = Array.from(allUsersMap.values());
+    for (const user of allUsers) {
+      emit(results.entity(location, user));
+    }
```

Comment:
Thinking about the error cases here, maybe it's a bit odd that you emit the users here outside the catch but groups are treated differently?

Maybe only wrap the things that can actually fail (like the `getOrganization*` functions) in a try/catch and abort entirely if that fails? And then always wire up and emit both users and groups if it didn't fail

Reply:
My thinking here was that if processing a particular GitHub org fails for any reason, it would log the failure but continue to process the remaining orgs which is why I wrapped the body of the `for` loop in a try/catch to continue the iteration in case of errors. 

As it is currently neither any users or groups from the particular org would be emitted if either of the `getOrgranization*` functions throws since `allUsersMap` would not be updated.